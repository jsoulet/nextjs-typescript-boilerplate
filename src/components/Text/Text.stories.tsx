import React from 'react'
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0'

import Text, { TextProps } from './index'

export default {
  title: 'Core/Text',
  component: Text,
} as Meta

const Template: Story<TextProps> = (args) => <Text {...args} />

export const Main = Template.bind({})
Main.args = {
  as: 'div',
  children: `You may say that I'm a dreamer, but I'm not the only one`,
}
