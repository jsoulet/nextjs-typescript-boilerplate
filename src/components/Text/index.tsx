/** @jsx jsx */
import React from 'react'
import { jsx } from '@emotion/core'
import tokens from '../tokens'
export interface TextProps {
  // eslint-disable-next-line no-undef
  as?: keyof JSX.IntrinsicElements
}

const Heading: React.FC<TextProps> = ({ as: As = 'div', children, ...props }) => {
  return (
    <As css={{ color: tokens.colors.black }} {...props}>
      {children}
    </As>
  )
}

export default Heading
