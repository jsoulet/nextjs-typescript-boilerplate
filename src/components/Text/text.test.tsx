import { render } from '@testing-library/react'
import { Main } from './Text.stories'

describe('Text', () => {
  it('matches snapshot', () => {
    const { container } = render(<Main>{`You may say that I'm a dreamer`}</Main>)
    expect(container).toMatchSnapshot()
  })
  it('renders children', () => {
    const { getByTestId, getByText } = render(
      <Main data-testid="text-id">{`But I'm not the only one`}</Main>
    )
    expect(getByTestId('text-id')).toBeVisible()
    expect(getByText(`But I'm not the only one`)).toBeVisible()
  })
})
