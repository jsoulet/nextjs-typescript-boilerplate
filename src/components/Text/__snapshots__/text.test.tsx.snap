// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text matches snapshot 1`] = `
<div>
  <div
    class="css-84mwxq-Heading"
  >
    You may say that I'm a dreamer
  </div>
</div>
`;
